server.py

```diff
@@ -21,8 +21,13 @@ def do_POST(self: Self) -> None:
 
       data = loads(self.rfile.read(int(self.headers['Content-Length'])))
 
+      if not data.get('state') or not data.get('params'):
+         self.respond(200, "No state or params found in the request body. Skipping provider state setup.")
+         return
+
       if (matches := graphql_query_state_regex.fullmatch(data['state'])) is None:
-        raise Exception(f'''State not supported: {data['state']}''')
+         self.respond(200, f'''State not supported: {data['state']} - skipping provider state setup.''')

```

Comment:
Please quote `data['state']` to distinguish it from the rest of the error message, for example `f'''State not supported: "{data['state']}" - skipping provider state setup.'''`

No concerns with double quotes here as they are not used as Python string delimiters but as literal separators in a string